packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -939,7 +940,8 @@ class PopupMenuButton<T> extends StatefulWidget {
        assert(offset != null),
        assert(enabled != null),
        assert(captureInheritedThemes != null),
-       assert(!(child != null && icon != null)), // fails if passed both parameters
+       assert(!(child != null && icon != null),
+           'You can only pass [child] or [icon], not both.'),
```

Comment:
Potentially this is simply personal preference. If you want, you can just revert this if you are a maintainer.

Reply:
I believe this makes sense. It would necessarily be a breaking change since an assert statement was added (I would imagine some developers might have mistakenly defined both and this would break their application upon upgrade)

Reply:
Also, it might be worth adding a regression test ensuring that an error is thrown when both have been defined.

Reply:
It is not a **new** `assert`, sorry for the confusion. I only moved the description of the assertion from a comment into the `assert` call.